onnxruntime/core/providers/cuda/tensor/upsample_impl.cu

```diff
@@ -161,25 +160,25 @@ void UpampleImpl(cudaStream_t stream,
   int blocksPerGrid = (int)(ceil(static_cast<float>(N) / GridDim::maxThreadsPerBlock));
   if (onnxruntime::UpsampleMode::NN == upsample_mode) {
     if (rank == 4) {
-      _UpampleNearestKernel<T,4><<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0, stream>>>(
+      _UpampleNearestKernel<T, 4><<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0, stream>>>(
           input_pitches, output_div_pitches, scales_div,
           input_data, output_data, N);
     } else if (rank == 3) {
-      _UpampleNearestKernel<T,3><<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0, stream>>>(
+      _UpampleNearestKernel<T, 3><<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0, stream>>>(
           input_pitches, output_div_pitches, scales_div,
           input_data, output_data, N);
     } else if (rank == 2) {
-      _UpampleNearestKernel<T,2><<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0, stream>>>(
+      _UpampleNearestKernel<T, 2><<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0, stream>>>(
           input_pitches, output_div_pitches, scales_div,
           input_data, output_data, N);
     } else if (rank == 1) {
-      _UpampleNearestKernel<T,1><<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0, stream>>>(
+      _UpampleNearestKernel<T, 1><<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0, stream>>>(
           input_pitches, output_div_pitches, scales_div,
           input_data, output_data, N);
     } else {
-      ORT_THROW("Unsupported rank by the Upsample CUDA kernel");
+      ORT_THROW("Unsupported rank by the Upsample CUDA kernel. Input rank: ", rank);
     }
-  } else if (onnxruntime::UpsampleMode::LINEAR) {
+  } else if (onnxruntime::UpsampleMode::LINEAR == upsample_mode) {
```

Comment:
Fixing this awkwardness. Technically, it will work fine for "correct" models as Upsample only supported Nearest and Linear modes. If it isn't Linear, it had to be Linear (for legit models). That explains why tests didn't break. Still good to fix the awkwardness.